torch/_inductor/kernel/mm_common.py

```diff
@@ -388,7 +388,7 @@ def filtered_configs(
 )
 
 # On ROCm convert num_stages to improve performance
-if torch.version.hip:
+if torch.version.hip and torch.cuda.is_available():
```

Comment:
At least some time ago `torch.cuda.is_available()` itself was leading to CUDA initialization: 
- https://github.com/pytorch/pytorch/issues/52675

Reply:
Thanks @vadimkantorov, let me give this a try locally.

Reply:
@vadimkantorov This no longer seems to be an issue for me

Triton backend initializing cuda:
```
import torch
import triton
driver = triton.runtime.driver
target = driver.active.get_current_target()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/triton/backends/amd/driver.py", line 497, in get_current_target
    device = self.get_current_device()
  File "/usr/local/lib/python3.10/dist-packages/torch/cuda/__init__.py", line 955, in current_device
    _lazy_init()
  File "/usr/local/lib/python3.10/dist-packages/torch/cuda/__init__.py", line 320, in _lazy_init
    torch._C._cuda_init()
RuntimeError: No HIP GPUs are available
```

cuda.is_available test:
```
torch.version.hip
'6.2.41134-65d174c3e'
torch.cuda.is_available()
False
```